enhancements/ingress/set-delete-http-headers.md

```diff
@@ -732,32 +732,37 @@ We don't allow setting the `strict-transport-security` header on either an Ingre
 #### X-Forwarded-For
 `x-forwarded-for` and related headers were considered for the deny-list (as modifying these headers using this new API could affect configuration specified using
 another [option](https://docs.openshift.com/container-platform/4.12/networking/ingress-operator.html#nw-using-ingress-forwarded_configuring-ingress))
-but have actual use-cases (which aren't covered by the existing option). They will be allowed, with notes in the API and product documentation,
-and possibly a warning on admission.
+but have actual use-cases (which aren't covered by the existing option). Therefore, they will be allowed, with the following note in the product documentation:
 
-For example: If someone creates an Ingress Controller with the following:
-```sh
-apiVersion: operator.openshift.io/v1
-kind: IngressController
-metadata:
-  name: default
-  namespace: openshift-ingress-operator
-spec:
-  httpHeaders:
-    forwardedHeaderPolicy: Replace
-```
-Or if someone add a route annotation `haproxy.router.openshift.io/set-forwarded-headers`,
-then the `set-header` stanzas will be added in `haproxy.config` using the [haproxy-config.template](https://github.com/openshift/router/blob/0cd9a58dff77127562f7851ccfa5cda79b8cef48/images/router/haproxy/conf/haproxy-config.template#L565-L580)
+Note: In the IngressController, if both the action `Set` is configured using `spec.httpHeaders.actions` and the 
+action `Replace` is configured using `spec.httpHeaders.forwardedHeaderPolicy`, then the resulting configured value
+is from `spec.httpHeaders.actions`.  
+In short, the value set by `spec.httpHeaders.actions` will override the value set by `spec.httpHeaders.forwardedHeaderPolicy`.
 
-However, if someone wants to set custom values for these then they need to use the API of this EP.
+In the case of HTTP request headers, the actions specified in `spec.httpHeaders.actions` on the Route will be executed after
+the actions specified in the IngressController's `spec.httpHeaders.actions` field.
 
-The custom values set using `spec.httpHeaders.actions` will take over the value defined in annotation `haproxy.router.openshift.io/set-forwarded-headers` or `spec.httpHeaders.forwardedHeaderPolicy`.  
+So, the custom HTTP request value of `x-forwarded-for` header set using `spec.httpHeaders.actions[*].request` of the Route will take over the value defined in annotation `haproxy.router.openshift.io/set-forwarded-headers` of Route, `spec.httpHeaders.forwardedHeaderPolicy` of Ingress Controller
+or `spec.httpHeaders.actions[*].request` of Ingress Controller as in case of HTTP request headers, HAProxy overrides the value of the frontend of same header name set by IngressController with the
+value which was present in the backend section.
```

Comment:
This sentence needs to be rewrittten.  Maybe make two sentences.  I can't make any sense of this.

Reply:
I have removed this as now we have a chart.